lib/glossarist/designation/expression.rb

```diff
@@ -5,15 +5,38 @@
 module Glossarist
   module Designation
     class Expression < Base
-      attr_accessor :gender
-      attr_accessor :part_of_speech
-      attr_accessor :plurality
       attr_accessor :prefix
       attr_accessor :usage_info
 
       # List of grammar info.
       # @return [Array<GrammarInfo>]
-      attr_accessor :grammar_info
+      attr_reader :grammar_info
+
+      def grammar_info=(grammar_info)
+        @grammar_info ||= []
+
+        grammar_info.each do |info|
+          @grammar_info << GrammarInfo.new(info)
```

Comment:
What happens in:
```ruby
x = Expression.new
x.grammar_info = [x,y,z] # assuming x y z are things GrammarInfo consumes
x.grammar_info = [x]
```
?

Reply:
Good catch. 
I have updated the PR.

Reply:
@HassanAkbar thanks. In this case do you just want to use `map`? i.e.
```ruby
grammar_info.map do |g|
  GrammarInfo.new(g)
end
```

Reply:
Yes, this is much better. Thank you.
I’ve updated the code.